src/ADAL.PCL/Internal-Telemetry/DefaultEvent.cs

```diff
@@ -0,0 +1,139 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+
+namespace Microsoft.IdentityModel.Clients.ActiveDirectory
+{
+    internal class DefaultEvent : EventsBase
+    {
+        internal static List<Tuple<string, string>> DefaultEvents = new List<Tuple<string, string>>();
+
+        static DefaultEvent()
+        {
+            ApplicationName = PlatformPlugin.PlatformInformation.GetApplicationName();
+
+            ApplicationVersion = PlatformPlugin.PlatformInformation.GetApplicationVersion();
+
+            SdkVersion = AdalIdHelper.GetAdalVersion();
+
+            SdkId = AdalIdHelper.GetAssemblyFileVersion();
+
+            DeviceId = PlatformPlugin.PlatformInformation.GetDeviceModel();
+        }
+
+        internal DefaultEvent(string eventName)
+        {
+            SetEvent(EventConstants.EventName,eventName);
+            //Fill in the default parameters
+
+            SetEvent(EventConstants.ApplicationName,ApplicationName);
```

Comment:
Not able to get what you mean...can you explain ?

---
In reply to: [74547305](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/494#discussion_r74547305) [](ancestors = 74547305)